lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -5446,8 +5446,89 @@ LogicalResult ConvertAtenOp<AtenEmptyMemoryFormatOp>::matchAndRewrite(
 
     rewriter.replaceOpWithNewOp<tosa::ConstOp>(op, resultType, emptyVal);
     return success();
+}
+
+// Calculates the min and max value of op that are observable.
+static LogicalResult inferNecessaryRange(Operation *op,
+                                         ConversionPatternRewriter &rewriter,
+                                         int64_t *min, int64_t *max) {
+    if (!op->hasOneUse()) {
+      return rewriter.notifyMatchFailure(
+          op, "can only convert round used by a single clamp");
+    }
+    Operation *user = *op->user_begin();
+
+    if (auto clamp = dyn_cast<AtenClampOp>(user)) {
+      if (!matchPattern(clamp.getMin(), m_TorchConstantInt(min)) ||
+          !matchPattern(clamp.getMax(), m_TorchConstantInt(max))) {
+        return rewriter.notifyMatchFailure(
+            op, "Clamp needs constant integer bounds");
+      }
+      return success();
+    }
+
+    if (auto add = dyn_cast<AtenAddScalarOp>(user)) {
+      int64_t alpha;
+      if (!matchPattern(add.getAlpha(), m_TorchConstantInt(&alpha)) ||
+          alpha != 1) {
+        return rewriter.notifyMatchFailure(
+            op, "Only add with alpha = 1 is handled");
+      }
+
+      int64_t summand;
+      if (!matchPattern(add.getOther(), m_TorchConstantInt(&summand))) {
+        return rewriter.notifyMatchFailure(
+            op, "Add needs constant integer second operand");
+      }
+
+      if (failed(inferNecessaryRange(add, rewriter, min, max)))
+        return failure();
+
+      // The IR looks like
+      //   a = x + summand
+      //   y = clamp a, min, max
+      // so x must have a dynamic range of [min - summand, max - summand] to
```

Comment:
Should this be `y`? Could say something like "`y` has a dynamic range of ...." as this is not describing a condition, but rather explaining a fact.

Reply:
I updated this to say
```
      // The IR looks like
      //   a = x + summand
      //   y = clamp a, min, max
      // which is the same as
      //   c = clamp a, min - summand, max - summand
      //   y = c + summand.
```